--- ../src-base/minecraft/org/bukkit/Bukkit.java
+++ ../src-work/minecraft/org/bukkit/Bukkit.java
@@ -301,6 +301,7 @@
      */
     public static void reload() {
         server.reload();
+        org.spigotmc.CustomTimingsHandler.reload(); // Spigot
     }
 
     /**
@@ -740,6 +741,15 @@
         return server.getIdleTimeout();
     }
 
+	/**
+	 * Checks if the server is in the process of being shutdown.
+	 *
+	 * @return true if server is in the process of being shutdown
+	 */
+	public static boolean isStopping() {
+		return server.isStopping();
+	}
+
     /**
      * @see Server#getUnsafe()
      */
