--- ../src-base/minecraft/org/bukkit/Server.java
+++ ../src-work/minecraft/org/bukkit/Server.java
@@ -12,11 +12,13 @@
 import java.util.UUID;
 import java.util.logging.Logger;
 
+import net.md_5.bungee.api.chat.BaseComponent;
 import org.bukkit.Warning.WarningState;
 import org.bukkit.command.CommandException;
 import org.bukkit.command.CommandSender;
 import org.bukkit.command.ConsoleCommandSender;
 import org.bukkit.command.PluginCommand;
+import org.bukkit.configuration.file.YamlConfiguration;
 import org.bukkit.entity.Entity;
 import org.bukkit.entity.Player;
 import org.bukkit.event.inventory.InventoryType;
@@ -911,9 +913,32 @@
      */
     public int getIdleTimeout();
 
+	/**
+	 * Checks if the server is in the process of being shutdown.
+	 *
+	 * @return true if server is in the process of being shutdown
+	 */
+	boolean isStopping();
+
     /**
      * @see UnsafeValues
      */
     @Deprecated
     UnsafeValues getUnsafe();
+
+    public Spigot spigot();
+
+    public static class Spigot {
+        public YamlConfiguration getConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        public void broadcast(BaseComponent component) {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+
+        public /* varargs */ void broadcast(BaseComponent ... components) {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+    }
 }
